tiledb/sm/query/query.cc

```diff
@@ -1605,7 +1609,10 @@ Status Query::set_condition(const QueryCondition& condition) {
         "to write queries"));
   }
 
-  condition_ = condition;
+  if (!condition.empty()) {
```

Comment:
We should actually throw an exception here instead of silently do nothing. This would also prevent users from doing anything funky... Like set a condition to something, then set it to an empty object expecting that it it is now cleared.

Reply:
That's fair. There was only one test that was doing funky things and from my light skimming it was specifically a "what if a user does this?" API QC type test.